pkg/prometheus/promcfg.go

```diff
@@ -215,6 +216,31 @@ func buildExternalLabels(p *v1.Prometheus) yaml.MapSlice {
 	return stringMapToMapSlice(m)
 }
 
+// ValidateConfigInputs validates if inputs passed are valid
+func (cg *ConfigGenerator) ValidateConfigInputs(p *v1.Prometheus) error {
+	logger := log.With(cg.logger, "Validating config inputs", "EnforcedBodySizeLimit")
+	if p.Spec.EnforcedBodySizeLimit != "" {
+		err := validateBodySizeLimit(p.Spec.EnforcedBodySizeLimit, logger)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// Validate EnforcedBodySizeLimit
+func validateBodySizeLimit(enforcedLimit string, logger log.Logger) error {
+	// To validate if given value is parsable for the acceptable body_size_limit values
+	_, err := units.ParseBase2Bytes(enforcedLimit)
+
+	if err != nil {
+		level.Warn(logger).Log("msg", "body_size_limit provided is invalid it should be a format of size",
+			"enforcedBodySizeLimit", enforcedLimit)
+		return errors.Wrap(err, "Invalid body_size_limit value specified")
```

Comment:
In general you should either return an error or log a message but not doing both.
See "Only handle errors once" in https://dave.cheney.net/2016/04/27/dont-just-check-errors-handle-them-gracefully. And the rest of the post is only worth reading ;-)

```suggestion
		return errors.Wrap(err, "invalid enforcedBodySizeLimit value specified")
```